src/main/java/com/example/booksstore/service/impl/BookServiceImpl.java

```diff
@@ -36,4 +36,19 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto book) {
+        if (bookRepository.existsById(id)) {
+            Book updatedBook = bookMapper.toModel(book);
+            updatedBook.setId(id);
+            return bookMapper.toDto(bookRepository.save(updatedBook));
+        }
+        throw new EntityNotFoundException("Can't find book by id " + id);
```

Comment:
let's flip this `if` the other way around to have ex throwing inside it's body instead of the main business logic to follow up with the return/fail early pattern